src/main/java/com/powsybl/openloadflow/dc/DcLoadFlowEngine.java

```diff
@@ -162,18 +183,32 @@ public DcLoadFlowResult run() {
         EquationSystem<DcVariableType, DcEquationType> equationSystem = context.getEquationSystem();
         DcLoadFlowParameters parameters = context.getParameters();
         TargetVector<DcVariableType, DcEquationType> targetVector = context.getTargetVector();
+        RunningContext runningContext = new RunningContext();
 
         // outer loop initialization
-        DcIncrementalPhaseControlOuterLoop phaseShifterControlOuterLoop = new DcIncrementalPhaseControlOuterLoop();
-        DcOuterLoopContext outerLoopContext = new DcOuterLoopContext(network);
+        List<Pair<DcOuterLoop, DcOuterLoopContext>> outerLoopsAndContexts = new ArrayList<>();
+
         if (parameters.getNetworkParameters().isPhaseControl()) {
-            phaseShifterControlOuterLoop.initialize(outerLoopContext);
+            DcIncrementalPhaseControlOuterLoop phaseShifterControlOuterLoop = new DcIncrementalPhaseControlOuterLoop();
+            DcOuterLoopContext phaseShifterControlOuterLoopContext = new DcOuterLoopContext(network);
+            outerLoopsAndContexts.add(Pair.of(phaseShifterControlOuterLoop, phaseShifterControlOuterLoopContext));
+            phaseShifterControlOuterLoop.initialize(phaseShifterControlOuterLoopContext);
+        }
+
+        if (parameters.isAreaInterchangeControl() && network.hasArea()) {
+            ActivePowerDistribution activePowerDistribution = ActivePowerDistribution.create(parameters.getBalanceType(), false, parameters.getNetworkParameters().isUseActiveLimits());
+            DcAreaInterchangeControlControlOuterLoop areaInterchangeControlOuterLoop = new DcAreaInterchangeControlControlOuterLoop(activePowerDistribution, parameters.getSlackBusPMaxMismatch(), parameters.getAreaInterchangePMaxMismatch());
+            DcOuterLoopContext areaInterchangeControlOuterLoopContext = new DcOuterLoopContext(network);
+            outerLoopsAndContexts.add(Pair.of(areaInterchangeControlOuterLoop, areaInterchangeControlOuterLoopContext));
+            areaInterchangeControlOuterLoop.initialize(areaInterchangeControlOuterLoopContext);
         }
 
         initStateVector(network, equationSystem, new UniformValueVoltageInitializer());
 
-        if (parameters.isDistributedSlack()) {
-            distributeSlack(network, network.getBuses(), parameters.getBalanceType(), parameters.getNetworkParameters().isUseActiveLimits());
+        double slackBusActivePowerMismatch = getActivePowerMismatch(network.getBuses());
+        double distributedActivePower = 0.0;
+        if (parameters.isDistributedSlack() || parameters.isAreaInterchangeControl()) {
+            distributedActivePower = distributeSlack(network, network.getBuses(), parameters.getBalanceType(), parameters.getNetworkParameters().isUseActiveLimits());
         }
```

Comment:
It was like this before, but here we do not account for slack distribution failure behavior parameter.
Could be in another PR, in this case just add a `// FIXME`.
Important note: in the case of AIC, failing here may not be relevant. Here is done only global slack distribution. But we may have the case of distribution failure when distributing mismatch globally, and at the same time no failure when distributing per area later on in the outerloops.